services/galley/src/Galley/API/Util.hs

```diff
@@ -370,7 +374,140 @@ runFederated remoteDomain rpc = do
   runExceptT (executeFederated remoteDomain rpc)
     >>= either (throwM . federationErrorToWai) pure
 
--- | Notify remote users of being added to a conversation
+-- | Convert an internal conversation representation 'Data.Conversation' to
+-- 'RegisterConversation' to be sent over the wire to a remote backend that will
+-- reconstruct this into multiple public-facing
+-- 'Wire.API.Conversation.Convevrsation' values, one per user from that remote
+-- backend.
+--
+-- FUTUREWORK: Include the team ID as well once it becomes qualified.
+toRegisterConversation ::
+  -- | The time stamp the conversation was created at
+  UTCTime ->
+  -- | The domain of the user that created the conversation
+  Domain ->
+  -- | The conversation to convert for sending to a remote Galley
+  Data.Conversation ->
+  -- | The resulting information to be sent to a remote Galley
+  RegisterConversation
+toRegisterConversation now localDomain Data.Conversation {..} =
+  MkRegisterConversation
+    { rcTime = now,
+      rcOrigUserId = Qualified convCreator localDomain,
+      rcCnvId = Qualified convId localDomain,
+      rcCnvType = convType,
+      rcCnvAccess = convAccess,
+      rcCnvAccessRole = convAccessRole,
+      rcCnvName = convName,
+      rcMembers = toMembers convLocalMembers convRemoteMembers,
+      rcMessageTimer = convMessageTimer,
+      rcReceiptMode = convReceiptMode
+    }
+  where
+    toMembers ::
+      [LocalMember] ->
+      [RemoteMember] ->
+      Map Domain [Member.Member]
+    toMembers ls rs =
+      let locals = Map.singleton localDomain . fmap localToMember $ ls
+          remotesUngrouped = fmap (second pure . remoteToMember) rs
+       in foldl' (flip (uncurry (Map.insertWith (<>)))) locals remotesUngrouped
+    localToMember :: LocalMember -> Member.Member
+    localToMember Members.InternalMember {..} =
+      Member.Member
+        { memId = memId,
+          ..
+        }
+    remoteToMember :: RemoteMember -> (Domain, Member.Member)
+    remoteToMember RemoteMember {..} =
+      ( qDomain . unTagged $ rmId,
+        Member.Member
+          { memId = qUnqualified . unTagged $ rmId,
+            memService = Nothing,
+            memOtrMuted = False,
+            memOtrMutedStatus = Nothing,
+            memOtrMutedRef = Nothing,
+            memOtrArchived = False,
+            memOtrArchivedRef = Nothing,
+            memHidden = False,
+            memHiddenRef = Nothing,
```

Comment:
This seems wrong, all of this data should be owned by user owning backend and not the conversation owning backend. So, it should never flow between domains.

Reply:
This has been taken care of in PR #1622.